src/Microsoft.ML.LightGBM/LightGbmMulticlassTrainer.cs

```diff
@@ -130,9 +131,9 @@ protected override void ConvertNaNLabels(IChannel ch, RoleMappedData data, float
         protected override void GetDefaultParameters(IChannel ch, int numRow, bool hasCategorical, int totalCats, bool hiddenMsg=false)
         {
             base.GetDefaultParameters(ch, numRow, hasCategorical, totalCats, true);
-            int numLeaves = int.Parse(Options["num_leaves"]);
+            int numLeaves = int.Parse(string.Format(CultureInfo.InvariantCulture, "{0}", Options["num_leaves"]));
```

Comment:
>int.Parse(string.Format(CultureInfo.InvariantCulture, "{0}", Options["num_leaves"])); [](start = 28, length = 85)

why can you just call int.Parse(Options["num_leaves"],CultureInfo.InvariantCulture); ?

Reply:
That won't work. You will need to convert object to string first and even then what advantage does this give over what I have now? I'm also concerned that we will convert to string without invariant and then try to parse with invariant flag.

---
In reply to: [199603671](https://github.com/dotnet/machinelearning/pull/454#discussion_r199603671) [](ancestors = 199603671)

Reply:
It would work if your Options["num_leaves"] was Invarianted string. But in your case Option["num_leaves"] contains already int value which you hide behind object. So all you need to is just a cast.
int numLeaves = (int)Options["num_leaves"];

---
In reply to: [199605092](https://github.com/dotnet/machinelearning/pull/454#discussion_r199605092) [](ancestors = 199605092,199603671)

Reply:
That will work but here I don't want to make an assumption that num_leaves is int, what if tomorrow someone inserts num_leaves as string or some other type? I'd rather have a generic way to translate to invariant string and then parse. What do you think?

---
In reply to: [199607828](https://github.com/dotnet/machinelearning/pull/454#discussion_r199607828) [](ancestors = 199607828,199605092,199603671)

Reply:
I have a proposal which will benefit all of us and remove any concerns regarding culture.  Don't do parsing at all. Just set numLeaves as out parameter in the function and set it properly in base call.


---
In reply to: [199611982](https://github.com/dotnet/machinelearning/pull/454#discussion_r199611982) [](ancestors = 199611982,199607828,199605092,199603671)

Reply:
In this case casting to int is fine. I spoke with Tom and he mentioned we can make this hard assumptions if the datastructure/code is internal.

---
In reply to: [199630347](https://github.com/dotnet/machinelearning/pull/454#discussion_r199630347) [](ancestors = 199630347,199611982,199607828,199605092,199603671)